pallets/account/src/tests.rs

```diff
@@ -0,0 +1,248 @@
+				&& EvmAccounts::<TestRuntime>::contains_key(ALICE)
+		);
+    });
+}
+
+#[test]
+fn account_claim_should_not_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        // invald signature
+        assert_noop!(
+            Accounts::claim_evm_account(
+                RuntimeOrigin::signed(ALICE),
+                Accounts::eth_address(&bob_secret()),
+                claim_signature(&BOB, &bob_secret())
+            ),
+            Error::<TestRuntime>::InvalidSignature
+        );
+        assert_noop!(
+            Accounts::claim_evm_account(
+                RuntimeOrigin::signed(ALICE),
+                Accounts::eth_address(&bob_secret()),
+                claim_signature(&ALICE, &alice_secret())
+            ),
+            Error::<TestRuntime>::InvalidSignature
+        );
+
+        assert_ok!(Accounts::claim_evm_account(
+            RuntimeOrigin::signed(ALICE),
+            Accounts::eth_address(&alice_secret()),
+            claim_signature(&ALICE, &alice_secret())
+        ));
+        // AccountId already mapped
+        assert_noop!(
+            Accounts::claim_evm_account(
+                RuntimeOrigin::signed(ALICE),
+                Accounts::eth_address(&alice_secret()),
+                claim_signature(&ALICE, &alice_secret())
+            ),
+            Error::<TestRuntime>::AccountIdHasMapped
+        );
+        // eth address already mapped
+        assert_noop!(
+            Accounts::claim_evm_account(
+                RuntimeOrigin::signed(BOB),
+                Accounts::eth_address(&alice_secret()),
+                claim_signature(&ALICE, &alice_secret())
+            ),
+            Error::<TestRuntime>::EthAddressHasMapped
+        );
+    });
+}
+
+#[test]
+fn account_default_claim_works() {
+    ExtBuilder::default().build().execute_with(|| {
+        let alice_default_evm =
+            <TestRuntime as Config>::DefaultAccountMapping::into_h160(ALICE.into());
+
+        // claim default account
+        assert_ok!(Accounts::claim_default_evm_account(RuntimeOrigin::signed(
+            ALICE
+        )));
+        System::assert_last_event(RuntimeEvent::Accounts(crate::Event::AccountClaimed {
+            account_id: ALICE.clone(),
+            evm_address: alice_default_evm.clone(),
+        }));
+
+        // check AddressManager mapping works
+        assert_eq!(
+            <Accounts as AddressManager<_, _>>::to_address(&ALICE),
+            Some(alice_default_evm)
+        );
+        assert_eq!(
+            <Accounts as AddressManager<_, _>>::to_account_id(&alice_default_evm),
+            Some(ALICE)
+        );
+
+        // should not allow to claim afterwards
+        assert_noop!(
+            Accounts::claim_evm_account(
+                RuntimeOrigin::signed(ALICE),
+                Accounts::eth_address(&alice_secret()),
+                claim_signature(&ALICE, &alice_secret())
+            ),
+            Error::<TestRuntime>::AccountIdHasMapped
+        );
+    });
+}
```

Comment:
Add some negative tests too.

We want to see it's not possible to do e.g. a replay attack.

Reply:
Negative tests? like this - `account_claim_should_not_work()`?

> We want to see it's not possible to do e.g. a replay attack.

Replay is simply not possible, since
- It's not possible to claim again
- Even if someone frontruns, it will still connect the accounts that are in signature created and signed by user

Reply:
Those are fine, although I'd suggest the same thing mentioned in some previous PR reviews (not only yours) - break up tests so if they fail, it's relatively clear from the test name what happened. This is just a suggestion though.

I was more implying the test for the replay.
> Replay is simply not possible, since
I'm not saying it's possible now, but to add tests for it that really ensure it's not possible.
You want to test things like that since you never know what piece of code might change in the future or what not.

Tests are a great tool to check your code works now as expected, but are even more valuable for the long run since they keep on ensuring your code keeps working as you expect.

Reply:
Got it, I'll break down the tests and make them more through